Modules/System/Azure Blob Services API/src/ABSClientImpl.Codeunit.al

```diff
@@ -204,32 +204,32 @@ codeunit 9051 "ABS Client Impl."
         SourceInStream: InStream;
     begin
         if UploadIntoStream('', '', '', FileName, SourceInStream) then
-            ABSOperationResponse := PutBlobBlockBlobStream(Filename, SourceInStream, ABSOptionalParameters);
+            ABSOperationResponse := PutBlobBlockBlobStream(Filename, SourceInStream, '', ABSOptionalParameters);
 
         exit(ABSOperationResponse);
     end;
 
-    procedure PutBlobBlockBlobStream(BlobName: Text; var SourceInStream: InStream; ABSOptionalParameters: Codeunit "ABS Optional Parameters"): Codeunit "ABS Operation Response"
+    procedure PutBlobBlockBlobStream(BlobName: Text; var SourceInStream: InStream; ContentType: Text; ABSOptionalParameters: Codeunit "ABS Optional Parameters"): Codeunit "ABS Operation Response"
```

Comment:
@t-tarasiuk you must not add a parameter to an existing procedure. (This would be breaking change).
You would have to create a new one with this additional parameter.

My suggestion would be to add this ContentType to the codeunit `ABS Optional Parameters` by adding a setter and a getter method tho this codeunit.
Then you could simple use this codeunit that is already contained in all procedures.

Reply:
In terms of breaking: in this case the codeunit is _internal_ (so is the procedure), so not really breaking.
There's also the new overload for function in the public client codeunit, so all good.

In terms of design: as far as I remember, the mandatory parameters are to be actual procedure's parameter. Everything non-mandatory is to be placed in `ABS Optional Parameters`.

Reply:
@mazhelez you're right I'm still having some problems with the new file browser on github, since it does not open the file automatically - therefore I was looking a the wrong file. 

Well since the current approach by using one default content-type is working I would suggest to place this property in the `ABS Optional Parameters` since it isn't mandatory.

Reply:
Hey @mazhelez, @pri-kise - thank you for a quick turnaround.

The idea behind overloading the existing functions was to maintain consistency with function `AppendBlockText` in codeunit `ABS Blob Client` that have different overloads. But if it does not fit the design we can add a new function `PutBlobBlockBlob `with two overloads: for `InStream `and `Text`, but with mandatory `ContentType `parameter.

Placing ContentType into `ABS Optional Parameters` might be a bit confusing as we have a bunch of existing functions (`PutBlobPageBlob`, `PutBlobAppendBlob`, `AppendBlockText`) that have both `ContentType `and  `ABS Optional Parameters`. In this case it may be not so obvious which where to define the parameter and what priorities they have.